connectors/src/connectors/zendesk/lib/zendesk_api.ts

```diff
@@ -1,27 +1,165 @@
+import type { ModelId } from "@dust-tt/types";
 import type { Client } from "node-zendesk";
 import { createClient } from "node-zendesk";
 
+import type { ZendeskFetchedArticle } from "@connectors/connectors/zendesk/lib/node-zendesk-types";
+import { ExternalOAuthTokenError } from "@connectors/lib/error";
+import logger from "@connectors/logger/logger";
+import { ZendeskBrandResource } from "@connectors/resources/zendesk_resources";
+
+const ZENDESK_RATE_LIMIT_MAX_RETRIES = 5;
+const ZENDESK_RATE_LIMIT_TIMEOUT_SECONDS = 60;
+
 export function createZendeskClient({
-  token,
+  accessToken,
   subdomain,
 }: {
-  token: string;
+  accessToken: string;
   subdomain: string;
 }) {
-  return createClient({ oauth: true, token, subdomain });
+  return createClient({ oauth: true, token: accessToken, subdomain });
 }
 
-export async function changeZendeskClientSubdomain({
-  client,
-  brandId,
-}: {
-  client: Client;
-  brandId: number;
-}) {
-  // TODO(2024-10-29 aubin): in some cases the brand in db is available and can be used to retrieve the subdomain
+/**
+ * Returns a Zendesk client with the subdomain set to the one in the brand.
+ * Retrieves the brand from the database if it exists, fetches it from the Zendesk API otherwise.
+ */
+export async function changeZendeskClientSubdomain(
+  client: Client,
+  { connectorId, brandId }: { connectorId: ModelId; brandId: number }
+) {
+  client.config.subdomain = await getZendeskBrandSubdomain(client, {
+    connectorId,
+    brandId,
+  });
+  return client;
+}
+
+/**
+ * Retrieves a brand's subdomain from the database if it exists, fetches it from the Zendesk API otherwise.
+ */
+export async function getZendeskBrandSubdomain(
+  client: Client,
+  { connectorId, brandId }: { connectorId: ModelId; brandId: number }
+): Promise<string> {
+  const brandInDb = await ZendeskBrandResource.fetchByBrandId({
+    connectorId,
+    brandId,
+  });
+  if (brandInDb) {
+    return brandInDb.subdomain;
+  }
+
   const {
     result: { brand },
   } = await client.brand.show(brandId);
-  client.config.subdomain = brand.subdomain;
-  return client;
+  return brand.subdomain;
+}
+
+/**
+ * Handles rate limit responses from Zendesk API.
+ * Expects to find the header `Retry-After` in the response.
+ * https://developer.zendesk.com/api-reference/introduction/rate-limits/
+ * @returns true if the rate limit was handled and the request should be retried, false otherwise.
+ */
+async function handleZendeskRateLimit(response: Response): Promise<boolean> {
+  if (response.status === 429) {
+    const retryAfter = Math.max(
+      Number(response.headers.get("Retry-After")) || 1,
+      1
+    );
+    if (retryAfter > ZENDESK_RATE_LIMIT_TIMEOUT_SECONDS) {
+      logger.info(
+        { retryAfter },
+        `[Zendesk] Attempting to wait more than ${ZENDESK_RATE_LIMIT_TIMEOUT_SECONDS} s, aborting.`
+      );
+      throw new Error(
+        `Zendesk retry after larger than ${ZENDESK_RATE_LIMIT_TIMEOUT_SECONDS} s, aborting.`
```

Comment:
Why abort? Why not wait the actual retry after time? 
Want to discuss a bit around this rate limiting?